### PR TITLE
Switch to PureCSV for csv parsing

### PR DESCRIPTION
**Pull request by oilnam**
After a few weird bugs and corner cases I decided rolling my own csv parser was NOT a good idea after all :)

So here's a partial rewrite that uses this [neat library](https://github.com/melrief/PureCSV) instead. 

I added a `RawReport` case class where all the fields are Options to handle missing values, and I provide a set of defaults in the `apply` method for the `Report` class. 

This is needed because the CSV that comes from Apple contains loads of empty fields, and PureCSV throws exceptions when trying to cast them to strings and ints. So my strategy is to provide a very flexible `RawReport` where everything is optional and convert that to a `Report` class where I define my own defaults (converting empty values with empty strings or 0s).

This is also the reason why in `compareReportsDiff` I'm filtering out reports with an empty `articleId`; I'm using the _unsafe_ version of the library for a similar reason, given that I'm doing my own field validation.

### CODE REVIEW

File: process-reports/src/main/scala/com/gu/contentapi/client/Reports.scala [hunk 1]
```diff
@@ -1,7 +1,28 @@
 package com.gu.contentapi.client
 
+import purecsv.unsafe._
+import java.io.File
+
 object Reports {
 
+  case class RawReport(
+    article: Option[String],
```
Comment (LATaylor-guardian) [inline process-reports/src/main/scala/com/gu/contentapi/client/Reports.scala]: Why is everything an Option?

File: process-reports/src/main/scala/com/gu/contentapi/client/Reports.scala [hunk 2]
```diff
@@ -22,46 +43,45 @@ object Reports {
...
+        publisherArticleId = raw.publisherArticleId.getOrElse("")
       )
     }
   }
 
   def compareReportsDiff(lastReportFileName: String, secondToLastReportFileName: String): List[Report] = {
 
-    val lastOneLines = io.Source.fromFile(lastReportFileName).getLines().drop(1) // first line being the csv schema
-    val secondLastLines = io.Source.fromFile(secondToLastReportFileName).getLines().drop(1)
+    val lastOneReports = CSVReader[RawReport].readCSVFromFile(new File(lastReportFileName), skipHeader = true)
```
Comment (LATaylor-guardian) [inline process-reports/src/main/scala/com/gu/contentapi/client/Reports.scala]: This and the function below do the same thing, but on a different file. You could create a function to reduce this.

File: process-reports/src/main/scala/com/gu/contentapi/client/Reports.scala (hunk 1 above)
Comment (oilnam) [inline process-reports/src/main/scala/com/gu/contentapi/client/Reports.scala]: @LATaylor-guardian I expanded the PR's description to answer this

File: process-reports/src/main/scala/com/gu/contentapi/client/Reports.scala (hunk 2 above)
Comment (oilnam) [inline process-reports/src/main/scala/com/gu/contentapi/client/Reports.scala]: I thought about it but it was already a one-liner, although very long. Might be good for readability to give it a pretty name though.

**Comment by LATaylor-guardian** [comment]
👍

**Pull request merged**